### PR TITLE
Move all browser manipulation logic to the BrowserManipulationQueue…#671 part 1)

### PR DESCRIPTION
**Pull request by inikulin**
Addtional improvements:
- Moved diffs to legacy dir
- Updated eslint and fix linting errors
- Switched from uuid to shortId

\cc @AlexanderMoskovkin @helen-dikareva @AndreyBelym

### CODE REVIEW

**Comment by inikulin** [comment]
\cc @VasilyStrelyaev Please, check comments

File: Gulpfile.js [hunk 1]
```diff
@@ -140,34 +140,38 @@ gulp.task('clean', function () {
...
-            'src/client/ui/cursor',
-            'src/client/driver/**/*.js'
-        ])
-        .pipe(eslint())
-        .pipe(eslint.format())
-        .pipe(eslint.failAfterError());
-});
+gulp.task('lint', function () {
+    // TODO: eslint supports only node with version 4 or higher.
+    // Remove condition once we get rid of node 0.10 support.
```
Comment (VasilyStrelyaev) [inline Gulpfile.js]:
```
    // TODO: eslint supports node version 4 or higher.
    // Remove this condition once we get rid of node 0.10 support.
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
Seems like just unstable client test. Please, review.

**Comment by inikulin** [comment]
ping @helen-dikareva @AndreyBelym @AlexanderMoskovkin

File: test/functional/fixtures/api/es-next/client-function/testcafe-fixtures/client-fn-test.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ test('Dispatch', async () => {
 });
 
 test('Call with arguments', async () => {
-    const getElementText = ClientFunction((className, idx) => document.querySelectorAll('.' + className)[idx].textContent);
+    const getElementText = ClientFunction((className, idx) => document.querySelectorAll('.' +
```
Comment (helen-dikareva) [inline test/functional/fixtures/api/es-next/client-function/testcafe-fixtures/client-fn-test.js]: Should we add `{...}` for function in this case?

**Comment by helen-dikareva** [comment]
lgtm

File: test/functional/fixtures/api/es-next/client-function/testcafe-fixtures/client-fn-test.js (hunk 2 above)
Comment (inikulin) [inline test/functional/fixtures/api/es-next/client-function/testcafe-fixtures/client-fn-test.js]: Yeah

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
FPR
\cc @AndreyBelym

**Comment by AndreyBelym** [comment]
lgtm

**Pull request merged**